src/index.ts

```diff
@@ -155,9 +170,9 @@ export async function getIronSession(
     options.cookieOptions.maxAge = computeCookieMaxAge(options.ttl);
   }
 
-  const sealFromCookies = cookie.parse(req.headers.cookie || "")[
-    options.cookieName
-  ];
+  const sealFromCookies = (
+    "cookies" in req ? req.cookies : cookie.parse(req.headers.cookie || "")
```

Comment:
Why was it necessary to switch between req.cookies and req.headers.cookie? Can't we always use req.headers.cookie?

Reply:
(I guess there's no req.headers.cookie in middlewares?)

Reply:
Its syntax is like this inside middlewares: `req.headers.get('cookie')`